src/ORM/Association/BelongsToMany.php

```diff
@@ -510,7 +510,7 @@ protected function _appendNotMatching($query, $options)
         $subquery = $this->_appendJunctionJoin($subquery, $conditions);
 
         $query
-            ->andWhere(function ($exp) use ($subquery, $conds) {
+            ->where(function ($exp) use ($subquery, $conds) {
```

Comment:
Okay, I'll revert this change and add method annotation to `QueryInterface` instead. In a previous PR @markstory had suggested changing the method call https://github.com/cakephp/cakephp/pull/11063#discussion_r134104138. I'll revert that too :)